### PR TITLE
C Grader: clearer message when expected output has line breaks

### LINKED ISSUE #8469: c autograder expected output extra space
I'm seeing some extra space when using the C autograder. The "expected" output box has some extra space before "BEGIN OUTPUT" (which comes from my tests). The bottom box is what I expect.

image

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Resolves #8469. The idea originally was that, if a single string is expected:
```
Expected:
        string
````

If more than one string ("one of" if one is needed, "all of" if all are needed):
```
Expected one of:
        string1
        string2
        string3
```

The problem is that this isn't clear when the expected input has line breaks:

```
Expected one of:
        string1 line1
string1 line2
        string2 line1
string2 line2
string2 line3
        string3
```

So this is now changed to the following if any of the expected outputs has a line break:

```
Expected one of:

string1 line1
string1 line2

string2 line1
string2 line2
string2 line3

string3
```

Before: 
image

After: 
image

### CODE REVIEW

File: graders/c/cgrader/cgrader.py [hunk 1]
```diff
@@ -452,17 +452,21 @@ def compile_re(t):
...
-            msg = f"Expected{comment}:\n\t" + "\n\t".join(
-                f"\033[32m{t}\033[0m"
-                if highlight_matches and r.search(outcmp) is not None
-                else t
+            msg = f"Expected{comment}:\n-\t" + "\n-\t".join(
+                str(
+                    f"\033[32m{t}\033[0m"
+                    if highlight_matches and r.search(outcmp) is not None
+                    else t
+                ).replace("\n", "\n\t")
```
Comment (nwalters512) [inline graders/c/cgrader/cgrader.py:460]: Help me understand what's actually happening here? What is the leading dash and why are tabs being used - seems like that would make the unexpected whitespace problem worse?

Comment (jonatanschroeder) [reply graders/c/cgrader/cgrader.py:460]: Updated the original PR comment for clarity.

Comment (nwalters512) [reply graders/c/cgrader/cgrader.py:460]: Thanks for the extra description. I worry that the dashes are ambiguous, especially when only one output is present. When I was first looking at the example output and trying to parse the code, I thought perhaps that `-` was a diff-style line prefix indicating that the output was missing a line or something.

Looking specifically at the original linked issue, I worry that with this change, a student is going to think that either a) their output must start with a `-` or b) their output must have a leading tab on each line, or both.

Perhaps for the case of a single option, we don't show any item marker and don't include any tabs. It's less clear to me what to do with multiple options.

Comment (jonatanschroeder) [reply graders/c/cgrader/cgrader.py]: Switched to blank line (`\n\n`) when any expected input has a line break.

Comment (nwalters512) [reply graders/c/cgrader/cgrader.py]: Discussed with Jonatan offline. If instructors don't like the default message, they can provide their own with `msg="..."` in their call to `test_run`.

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**